README


This archive contains the distribution Mail-Chimp2,
version 0.4:

  Mailchimp V2 API

This software is copyright (c) 2013 by ideegeo Group Limited.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.008.



Changes
==================================================
Changes from 2013-02-03 00:00:00 +0000 to present.
==================================================

----------------------------------------
version 0.4 at 2014-02-03 23:54:32 +0000
----------------------------------------

  Change: 0b84e80feec970dab51dd541315867fd16069160
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2014-02-04 12:54:18 +0000

    push a latest branch that can be installed from 

  Change: 8aa81872540af108ade2f376634384c672d03710
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2014-02-04 12:48:53 +0000

    v0.3 

----------------------------------------
version 0.3 at 2014-02-03 23:48:16 +0000
----------------------------------------

  Change: 4481048dc7819ec3e7bdf20373b24d35e5b4f749
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2014-02-04 12:48:16 +0000

    make the DC autoselector a bit more robust 

  Change: 5e96bc384ca2c00deb292d7f51b47cfca03cc6f5
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2014-01-21 16:31:27 +0000

    parameter handling for importer 

  Change: ef3e35140903450339179532cfd4629319bedaea
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 15:16:05 +0000

    v0.2 

----------------------------------------
version 0.2 at 2013-12-30 02:15:42 +0000
----------------------------------------

  Change: bd37d11155ec96c2c55ce550a0e2adc00161a906
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 15:15:42 +0000

    dist.ini tuning 

----------------------------------------
version 0.1 at 2013-12-30 02:07:46 +0000
----------------------------------------

  Change: 19630a72c1477e1b9786ca4484a0b0de00cacfa0
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 15:07:46 +0000

    satisfy Perl::Critic 

  Change: 498d0366540c7f3d557472c4debffafca3d7c850
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 15:06:45 +0000

    satisfy Perl::Critic 

  Change: b6354b84b95bb94e2ee1960448314c7ff1afa84a
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 15:04:55 +0000

    satisfy Perl::Critic 

  Change: 7f8a66d467bb9446b13268c4af1e8571130edb09
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 14:57:36 +0000

    satisfy Perl::Critic 

  Change: 23e4952526c167598569a5ca66e092a3ed9afd30
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 14:51:07 +0000

    satisfy dzil 

  Change: f6f3c7b265d7afa904b6b761ee49019fe03a9edc
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 14:47:53 +0000

    added a basic CSV importer 

  Change: f0bd2c47282eaa94f0ceb8f735e234334e2bf40c
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 14:15:30 +0000

    added some docs 

  Change: 8fae91d16bda253699c299fff400e4962feb40a8
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 14:07:30 +0000

    first draft 

  Change: b556629d43cc462b7d12e99a54897b6ca898435b
  Author: Lenz Gschwendtner <lenz@springtimesoft.com>
  Date : 2013-12-30 12:42:38 +0000

    initial commit 

================
End of releases.
================


lib/Mail/Chimp2.pm
package Mail::Chimp2;

use 5.012;
use Mouse;

# ABSTRACT: Mailchimp V2 API

our $VERSION = '0.4'; # VERSION

with 'Web::API';
use Data::Dump;


has 'dc' => (
    is      => 'rw',
    isa     => 'Str',
    default => sub { 'us1' },
);

has 'commands' => (
    is      => 'rw',
    default => sub {
        {
            # campaign handling
            campaigns_list => { path => 'campaigns/list' },

            # list handling
            lists_list    => { path => 'lists/list' },
            lists_members => {
                path      => 'lists/members',
                mandatory => ['id']
            },
            lists_batch_subscribe => {
                path      => 'lists/batch-subscribe',
                mandatory => [ 'id', 'batch' ]
            },
            lists_subscribe => {
                path      => 'lists/subscribe',
                mandatory => [ 'id', 'email' ]
            },
            lists_unsubscribe => {
                path      => 'lists/unsubscribe',
                mandatory => [ 'id', 'email' ]
            },

        };
    });


sub commands {
    my ($self) = @_;
    return $self->commands;
}


sub BUILD {
    my ($self) = @_;

    my ($_key, $_dc) = split(/-/, $self->api_key);
    $self->dc($_dc) if $_dc;
    $self->user_agent(__PACKAGE__ . ' ' . $Mail::Chimp2::VERSION);
    $self->strict_ssl(1);
    $self->content_type('application/json');
    $self->default_method('POST');
    $self->extension('json');
    $self->base_url('https://' . $self->dc . '.api.mailchimp.com/2.0');
    $self->auth_type('hash_key');
    $self->api_key_field('apikey');

    return $self;
}


__PACKAGE__->meta->make_immutable();    # End of Mail::Chimp2

__END__

=pod

=encoding UTF-8

=head1 NAME

Mail::Chimp2 - Mailchimp V2 API

=head1 VERSION

version 0.4

=head1 SYNOPSIS

This is a very rudimentary start of a Mailchimp API version 2
integration. It will be a complete implementation eventually but for now
all I needed was bulk add addresses to lists, so that is what i covered
so far.

    use Mail::Chimp2;

    my $chimp = Mail::Chimp2->new(
        debug   => 1,
        api_key => 'yourapikey-us1'
    );
    dump $chimp->lists_batch_subscribe(
        id              => 'list_id',
        double_optin    => 'false',
        update_existing => 'true',
        batch           => [ {
                email      => { email => 'me@example.com' },
                merge_vars => { FNAME => 'Me', LNAME => 'Example' } 
        } ]
    );

=head1 ATTRIBUTES

=head2 dc

The datacenter attribute is set to the extension of the API key. You can force
it if you need by setting this attribute.

=head1 INTERNALS

=head2 BUILD

basic configuration for the client API happens usually in the BUILD method when using Web::API

=head1 BUGS

Please report any bugs or feature requests on GitHub's issue tracker L<https://github.com/norbu09/Mail::Chimp2/issues>.
Pull requests welcome.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Mail::Chimp2

You can also look for information at:

=over 4

=item * GitHub repository

L<https://github.com/norbu09/Mail::Chimp2>

=item * MetaCPAN

L<https://metacpan.org/module/Mail::Chimp2>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Mail::Chimp2>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Mail::Chimp2>

=back

=head1 ACKNOWLEDGEMENTS

=over 4

=back

=head1 AUTHOR

Lenz Gschwendtner <lenz@springtimesoft.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by ideegeo Group Limited.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.008.
use strict;
use warnings;

use 5.012;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Mailchimp V2 API",
  "AUTHOR" => "Lenz Gschwendtner <lenz\@springtimesoft.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Mail-Chimp2",
  "EXE_FILES" => [
    "bin/import.pl"
  ],
  "LICENSE" => "perl",
  "NAME" => "Mail::Chimp2",
  "PREREQ_PM" => {
    "Data::Dump" => 0,
    "Getopt::Long" => 0,
    "Mouse" => 0,
    "Text::CSV" => 0,
    "Web::API" => 0,
    "strict" => 0
  },
  "TEST_REQUIRES" => {},
  "VERSION" => "0.4",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Data::Dump" => 0,
  "Getopt::Long" => 0,
  "Mouse" => 0,
  "Text::CSV" => 0,
  "Web::API" => 0,
  "strict" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



